### ISSUE TITLE
Elasticsearch cross-cluster search

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#13609]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon Elasticsearch Service now offers support for cross-cluster search.

### New or Affected Resource(s)

* `aws_elasticsearch_cross_cluster_search_connection`
* `aws_elasticsearch_cross_cluster_search_connection_accepter`

### Potential Terraform Configuration

```hcl
resource "aws_elasticsearch_cross_cluster_search_connection" "example" {
  connection_alias = "..."

  source_domain_info {
  }

  destination_domain_info {
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/06/announcing-cross-cluster-search-support-for-amazon-elasticsearch-service/).
[Developer guide](https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/cross-cluster-search.html).

Requires [AWS SDK v1.31.10](https://github.com/aws/aws-sdk-go/releases/tag/v1.31.10):
* https://github.com/terraform-providers/terraform-provider-aws/pull/13459

**Comment by Symbianx** [comment]
@ewbankkit implemented support for this in PR #22988.

**Comment by Symbianx** [comment]
hey @ewbankkit, I understand there's a lot of issues and PR's going into this repo. Is there anything I can do to help in getting my PR looked at?
